### PR TITLE
Secure-Code-Warrior-Plugin Blog Post

### PR DESCRIPTION
**Pull request by eheimbuch**

### CODE REVIEW

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+Hey SCM-Manager community,
+
+we are proud to inform you about our newest (and first) official partnership with [Secure Code Warrior](https://www.securecodewarrior.com/).
+As part of our cooperation, we created the `SCM-SCW-Plugin` which integrates the Secure Code Warrior learning material into SCM-Manager.
+Beside GitHub SCM-Manager is the second exclusive integration with Secure Code Warrior support.
+
+## Who is Secure Code Warrior?
+[Secure Code Warrior](https://www.securecodewarrior.com/) offers a platform that provides developers, 
+as well as others interested in security, with knowledge about security vulnerabilities. 
+The approaches of using gamification and micro-learning for this work very well and thus manage
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:22]: Singular?
```suggestion
The approache of using gamification and micro-learning for this work very well and thus manage 
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+keywords:
+- cloudogu
+- secure-code-warrior
+---
+
+Hey SCM-Manager community,
+
+we are proud to inform you about our newest (and first) official partnership with [Secure Code Warrior](https://www.securecodewarrior.com/).
+As part of our cooperation, we created the `SCM-SCW-Plugin` which integrates the Secure Code Warrior learning material into SCM-Manager.
+Beside GitHub SCM-Manager is the second exclusive integration with Secure Code Warrior support.
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:17]:
```suggestion
Besides GitHub, SCM-Manager is the second exclusive integration with Secure Code Warrior support.
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Who is Secure Code Warrior?
+[Secure Code Warrior](https://www.securecodewarrior.com/) offers a platform that provides developers, 
+as well as others interested in security, with knowledge about security vulnerabilities. 
+The approaches of using gamification and micro-learning for this work very well and thus manage 
+to sensitize developers to the major topic of software security in the long term.
+
+## What is the SCM-SCW-Plugin?
+The `SCM-SCW plugin` brings parts of the `Secure Code Warrior` knowledge into your SCM manager.
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:26]:
```suggestion
The `SCM-SCW plugin` brings parts of the `Secure Code Warrior` knowledge into your SCM-Manager. 
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+The `SCM-SCW plugin` brings parts of the `Secure Code Warrior` knowledge into your SCM manager. 
+We focused mainly on two scenarios for the first implementation, 
+both of which depend on the pull requests you may know from the `SCM-Review-Plugin`.
+
+### Scenario 1: Pull Request Description
+
+A developer from our team has found a critical `SQLI` (`SQL Injection`) security bug in our code.
+He fixes the vulnerability on a new bugfix branch and creates a pull request in our SCM-Manager. 
+
+`Issue:`
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:35]: I don't like code blocks for emphases
```suggestion
**Issue:**
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+### Scenario 1: Pull Request Description
+
+A developer from our team has found a critical `SQLI` (`SQL Injection`) security bug in our code.
+He fixes the vulnerability on a new bugfix branch and creates a pull request in our SCM-Manager. 
+
+`Issue:` 
+Unfortunately, no other developer has knowledge about `SQLI` and therefore cannot verify his changes.
+
+`Solution:`
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:38]: I don't like code blocks for emphases
```suggestion
**Solution:**
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+As soon as the pull request contains keywords related to its security-related fix, such as `SQLI`, 
+in the title or description, `Secure Code Warrior` learning content will be shown directly on the pull request.
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:40]:
```suggestion
in the title or description, Secure Code Warrior learning content will be shown directly on the pull request.
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+Now the reviewers can learn about SQL injection before reviewing the changes.
+
+![Pull Request Description](assets/SCW-PR-Description.jpg)
+
+### Scenario 2: Pull Request Comments
+A developer from our team creates a new feature regarding sql statements. 
+He also pushes his feature on a new feature branch and creates a pull request in SCM-Manager.
+
+Afterwards a reviewer checks the new features and find some issues in the code.
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:49]:
```suggestion
Afterwards a reviewer checks the new features and finds some issues in the code. 
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+### Scenario 2: Pull Request Comments
+A developer from our team creates a new feature regarding sql statements. 
+He also pushes his feature on a new feature branch and creates a pull request in SCM-Manager.
+
+Afterwards a reviewer checks the new features and find some issues in the code. 
+There is an attack vector which can be exploited with SQL injection using user-controlled inputs.
+The reviewer creates pull request comments describing his findings.
+
+`Issue:`
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:53]:
```suggestion
**Issue:**
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+Now the developer needs to understand how this vulnerability works and how to fix it.
+
+`Solution:`
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:56]:
```suggestion
**Solution:**
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+As soon as the reviewer post pull request comments which contains special keywords like `sqli` or `SQL injection`,
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:57]:
```suggestion
As soon as the reviewer posts pull request comments which contain special keywords like `sqli` or `SQL injection`,
```

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+the Secure-Code-Warrior-Plugin post more comments underneath with matching content from `Secure Code Warrior`.
```
Comment (pfeuffer) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:58]:
```suggestion
the Secure-Code-Warrior-Plugin posts more comments underneath with matching content from Secure Code Warrior. 
```

**Review by pfeuffer — COMMENTED** [review]
Found some typos (at least I think so), and I would not recommend to use code blocks for highlighting. This looks strange in my eyes. Better use **bold** or *italic* font (but maybe that's just me).

File: content/posts/2021-06-15-scm-scw-plugin/index.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+Besides GitHub, SCM-Manager is the second exclusive integration with Secure Code Warrior support.
+
+## Who is Secure Code Warrior?
+[Secure Code Warrior](https://www.securecodewarrior.com/) offers a platform that provides developers, 
+as well as others interested in security, with knowledge about security vulnerabilities. 
+The approache of using gamification and micro-learning for this work very well and thus manage 
+to sensitize developers to the major topic of software security in the long term.
+
+## What is the SCM-SCW-Plugin?
+The `SCM-SCW plugin` brings parts of the `Secure Code Warrior` knowledge into your SCM-Manager.
```
Comment (eheimbuch) [inline content/posts/2021-06-15-scm-scw-plugin/index.md:26]:
```suggestion
The `SCM-SCW-Plugin` brings parts of the `Secure Code Warrior` knowledge into your SCM-Manager. 
```

**Pull request merged**